misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -89,29 +91,112 @@
         String foundTypeLeft = matcher.group(1);
         String foundTypeRight = matcher.group(2);
 
+        boolean foundErrorLeft = false;
+        boolean foundErrorRight = false;
+
+        // If the project has multiple modules
+        if (foundTypeLeft.contains(":")) {
+            foundTypeLeft = foundTypeLeft.split(":")[foundTypeLeft.split(":").length - 1];
+            // If the check or checkpanic is to get the symbol name
+            if (foundTypeLeft.contains("|")) {
+                foundTypeLeft = foundTypeLeft.split("[|]")[0];
+                foundErrorLeft = true;
+            }
+        }
+
+        // If the project has multiple modules
+        if (foundTypeRight.contains(":")) {
+            foundTypeRight = foundTypeRight.split(":")[foundTypeRight.split(":").length - 1];
+            // If the function is returning an error need to get the symbol name
+            if (foundTypeRight.contains("|")) {
+                foundTypeRight = foundTypeRight.split("[|]")[0];
+                foundErrorRight = true;
+            }
+        }
+
+        // If the check or checkpanic is to get the symbol name
+        if (foundTypeLeft.contains("|")) {
+            foundTypeLeft = foundTypeLeft.split("[(|]")[1];
+            foundErrorLeft = true;
+        }
+
+        // If the function is returning an error need to get the symbol name
+        if (foundTypeRight.contains("|")) {
+            foundTypeRight = foundTypeRight.split("[(|]")[1];
+            foundErrorRight = true;
+        }
+
         // Get the semantic model
         SemanticModel semanticModel = context.workspace().semanticModel(context.filePath()).orElseThrow();
 
         // To restrict the code action from appearing for handled cases.
         List<Symbol> fileContentSymbols = semanticModel.moduleLevelSymbols();
-        boolean foundLeft = fileContentSymbols.stream().anyMatch(p -> p.name().contains(foundTypeLeft));
-        boolean foundRight = fileContentSymbols.stream().anyMatch(p -> p.name().contains(foundTypeRight));
+        String finalFoundTypeLeft = foundTypeLeft;
+        boolean foundLeft = fileContentSymbols.stream().anyMatch(p -> p.name().contains(finalFoundTypeLeft));
+        String finalFoundTypeRight = foundTypeRight;
+        boolean foundRight = fileContentSymbols.stream().anyMatch(p -> p.name().contains(finalFoundTypeRight));
```

Comment:
Shall we remove this symbols matching logic with the diagnostic properties. With the latest changes you can retrive RHS symbol information through the `diagnostic.properties()` api. Have a look in [here](https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java#L67-L72). Since the sample code-action uses same incompatible types diagnostic you can retrieve the found symbol at index 1 (eg. `diagnostic.properties().get(1)`).

Reply:
Changed the code according to the comment. Used `diagnostic.properties()`